snakemake/script.py

```diff
@@ -821,6 +821,293 @@ def execute_script(self, fname, edit=False):
         self._execute_cmd("julia {fname:q}", fname=fname)
 
 
+class RustScript(ScriptBase):
+    @staticmethod
+    def generate_preamble(
+        path,
+        source,
+        basedir,
+        input_,
+        output,
+        params,
+        wildcards,
+        threads,
+        resources,
+        log,
+        config,
+        rulename,
+        conda_env,
+        container_img,
+        singularity_args,
+        env_modules,
+        bench_record,
+        jobid,
+        bench_iteration,
+        cleanup_scripts,
+        shadow_dir,
+        preamble_addendum="",
+    ):
+        wrapper_path = path[7:] if path.startswith("file://") else path
+
+        if not log:
+            log = ""
+        else:
+            lookup = {
+                (True, True, True): " >> {0} 2>&1",
+                (True, False, True): " >> {0}",
+                (False, True, True): " 2>> {0}",
+                (True, True, False): " > {0} 2>&1",
+                (True, False, False): " > {0}",
+                (False, True, False): " 2> {0}",
+            }
+            log = lookup[(stdout, stderr, append)].format(self.log)
```

Comment:
@tedil `self`, `stdout`, `stderr` and `append` don't exist in this context. I didn't want to change anything as I wasn't super clear on the intended functionality here.

Reply:
Ah I just copied the PythonScript class for reference, so there are still python specific tidbits in there.

Reply:
Ok, I'll address this in my next round of work.